### PR TITLE
Handle no leaderboard layout on public profile

### PR DESCRIPTION
**Pull request by mohitb35**
This handles the public profile layout when no leaderboard should be shown.

Changes:
1. When no leaderboard is shown, the My Contributions section takes the full width of the screen
2. When no leaderboard is shown, and space permits, the MyContributionsList can contain 2 columns (subject to a minimum card width of 450 px).

For testing, the leaderboard will always be hidden on the public profile.

### CODE REVIEW

**Review by mariahosfeld — COMMENTED** [review]
LGTM, tested on desktop and on mobile, works as expected.

File: src/features/user/MFV2/MyContributions/MyContributions.module.scss [hunk 1]
```diff
@@ -308,10 +312,14 @@
 .registrationItemCard {
   display: flex;
   flex-direction: column;
+  justify-content: center;
   background-color: #ffffff;
   border: 3px solid rgba(#007a49, 0.1);
   border-radius: 12px;
   overflow: hidden;
+  width: 100%;
+  min-width: 0; // Allow shrinking below 450px
```
Comment (prachigarg19) [inline src/features/user/MFV2/MyContributions/MyContributions.module.scss:321]: Why was this added? The registration cards seem to shrink in a similar way for width<450px without specifying `min-width:0`

**Review by prachigarg19 — APPROVED** [review]

File: src/features/user/MFV2/MyContributions/MyContributions.module.scss (hunk 1 above)
Comment (mohitb35) [reply src/features/user/MFV2/MyContributions/MyContributions.module.scss:321]: These styles apply to the project card and registration cards. When this wasn't done, the cards stopped resizing as the container got narrower, and the content at the right got hidden.
